File: Data Structures & Algorithms/sort-an-array/submission-3.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:

        if len(nums)<=1:
            return nums

        n=len(nums)

        mid=n//2

        first=self.sortArray(nums[:mid])
        last=self.sortArray(nums[mid:])

        def merge(first,second):

            i=0
            j=0
            res=[]

            while i<len(first) and j<len(last):
                if first[i]<=last[j]:
                    res.append(first[i])
                    i+=1
                else:
                    res.append(last[j])
                    j+=1

            while i<len(first):
                res.append(first[i])
                i+=1
            while j<len(last):
                res.append(last[j])
                j+=1

            return res

        return merge(first,last)


        
        

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {

        int k=0;

        for(int i=0; i<nums.Length; i++){
            if (nums[i]!=nums[k]){
                k++;
                nums[k]=nums[i];
                
                
            }
        }
        return k+1;
        
    }
}